src/main/java/com/onarandombox/MultiverseCore/utils/result/Result.java

```diff
@@ -0,0 +1,84 @@
+package com.onarandombox.MultiverseCore.utils.result;
+
+import java.util.NoSuchElementException;
+import java.util.function.Consumer;
+
+public sealed interface Result<S extends SuccessReason, F extends FailureReason> permits Result.Success, Result.Failure {
+    static <F extends FailureReason, S extends SuccessReason> Result<S, F> success(S successReason) {
+        return new Success<>(successReason);
+    }
+
+    static <F extends FailureReason, S extends SuccessReason> Result<S, F> failure(F failureReason) {
+        return new Failure<>(failureReason);
+    }
+
+    boolean isSuccess();
+
+    boolean isFailure();
+
+    S successReason();
+
+    F failureReason();
```

Comment:
Funny thing is, since this is a sealed interface, we could use default implementations that check `instanceof` exhaustively. Not really sure if there's anything better about either way though.